### PR TITLE
audio: make clock control optional in SOF Zephyr builds (plus remove clk platform code for TGL as example)

### PR DESCRIPTION
**Pull request by kv2019i**
This is first step in solving in simplifying clock management code in SOF Zephyr builds ( https://github.com/thesofproject/sof/issues/9541 ).

This PR does the following
 - isolate all uses of SOF-side clock framework (rtos/clk.h) in generic application code (there are not many!)
 - enable SOF Zephyr to be built without linking in rtos/clk.h, and use Intel cAVS2.5 as example (lot of unnecessary code dropped from the build!)
 - no changes yet to non-Intel platform code (seems doable espcially for @thesofproject/nxp  native Zephyr platforms), but didn't do this yet

The big elephant in the room is cpu-clk-manager.c. This is an optional feature, currently only used by Intel, but is also the only legitimate use of the clock interface. IOW, this code really needs to know the available CPU/DSP frequencies, be able to get the current frequency and set it. And ideally do this in a way that works on multiple platforms. 
**UPDATE:** some update on this, a clock management framework is under discussion in Zephyr that would meet the needs of SOF cpu-clk-manager, see more details in https://github.com/thesofproject/sof/issues/9541#issuecomment-2493761190

The next step for me would be to see how to reimplement cpu-clk-manager.c directly on top of Zephyr clock-control.h interface. There are parts available (clock-control.h and Intel "intel,adsp-shim-clkctl"" DT), but not all so this is likely to require Zephyr-side development.

### CODE REVIEW

File: src/platform/intel/ace/platform.c [hunk 1]
```diff
@@ -86,16 +86,15 @@ static struct pm_notifier pm_state_notifier = {
...
 
 	trace_point(TRACE_BOOT_PLATFORM_CLOCK);
 	platform_clock_init(sof);
-	kcps_budget_init();
 
 #if CONFIG_KCPS_DYNAMIC_CLOCK_CONTROL
+	kcps_budget_init();
+
 	ret = core_kcps_adjust(cpu_get_id(), PRIMARY_CORE_BASE_CPS_USAGE);
-#else
```
Comment (kv2019i) [inline src/platform/intel/ace/platform.c:97]: @abonislawski @serhiy-katsyuba-intel Can you double-check this? It seems unnecessary to set the DSP frequency from SOF side if there is no dynamic control (the Zephyr platform code should choose the default speed).

File: src/platform/intel/cavs/platform.c [hunk 2]
```diff
@@ -116,14 +116,11 @@ int platform_init(struct sof *sof)
...
 
-	trace_point(TRACE_BOOT_PLATFORM_CLOCK);
-	platform_clock_init(sof);
-
 	trace_point(TRACE_BOOT_PLATFORM_SCHED);
 	scheduler_init_edf();
 
 	/* init low latency timer domain and scheduler */
-	sof->platform_timer_domain = timer_domain_init(sof->platform_timer, PLATFORM_DEFAULT_CLOCK);
+	sof->platform_timer_domain = timer_domain_init(sof->platform_timer, 0);
```
Comment (kv2019i) [inline src/platform/intel/cavs/platform.c:123]: The "clk" parameter has no use on Zephyer, so might as well pass a 0.

File: src/audio/base_fw.c [hunk 3]
```diff
@@ -19,7 +19,7 @@
 #include <sof/lib_manager.h>
 #include <rtos/clk.h>
 #include <rtos/init.h>
-#include <platform/lib/clk.h>
+
```
Comment (kv2019i) [inline src/audio/base_fw.c:22]: Hmm, I thought I had a point in adding this empty lne for grouping, but now I realize I really don't. Will fix in next version.

**Review by kv2019i — COMMENTED** [review]
As this a more complicated PR, added some annotations inline to the code to help in review.

File: src/library_manager/lib_manager.c [hunk 4]
```diff
@@ -963,13 +963,15 @@ static int lib_manager_setup(uint32_t dma_id)
...
 
+#if CONFIG_KCPS_DYNAMIC_CLOCK_CONTROL
 	/*
 	 * make sure that the DSP is running full speed for the duration of
 	 * library loading
 	 */
 	ret = core_kcps_adjust(cpu_get_id(), CLK_MAX_CPU_HZ / 1000);
 	if (ret < 0)
 		goto err_dma_buffer;
+#endif
```
Comment (lyakh) [inline src/library_manager/lib_manager.c:974]: maybe a dummy inline `core_kcps_adjust()` implementation instead?

File: src/platform/intel/ace/platform.c [hunk 5]
```diff
@@ -86,16 +86,15 @@ static struct pm_notifier pm_state_notifier = {
...
-	ret = core_kcps_adjust(cpu_get_id(), CLK_MAX_CPU_HZ / 1000);
 #endif
 	if (ret < 0)
 		return ret;
```
Comment (lyakh) [inline src/platform/intel/ace/platform.c:100]: please, move this check under `#ifdef` and remove `ret` initialisation

File: src/platform/intel/cavs/platform.c (hunk 2 above)
Comment (lyakh) [reply src/platform/intel/cavs/platform.c:123]: deserves a comment?

File: src/library_manager/lib_manager.c (hunk 4 above)
Comment (kv2019i) [reply src/library_manager/lib_manager.c:974]: @lyakh Tried this approach, but it gets a bit ugly as CLK_MAX_CPU_HZ is not defined either. Or we want to drop the requirement to define such values on SOF side.

File: src/platform/intel/ace/platform.c (hunk 5 above)
Comment (kv2019i) [reply src/platform/intel/ace/platform.c:99]: Done in today's PR revision.

File: src/platform/intel/cavs/platform.c (hunk 2 above)
Comment (kv2019i) [reply src/platform/intel/cavs/platform.c:123]: Done in today's PR revision.

File: src/platform/intel/ace/platform.c (hunk 1 above)
Comment (kv2019i) [reply src/platform/intel/ace/platform.c:97]: @abonislawski any comment?

**Review by lgirdwood — APPROVED** [review]
Nice job @kv2019i - more legacy being removed .
@dbaluta some good code savings as we purge more legacy

**Review by lyakh — APPROVED** [review]

**Pull request merged**